plonky2/src/hash/hash_types.rs

```diff
@@ -31,14 +31,12 @@ impl<F: Field> HashOut<F> {
         }
     }
 
-    pub fn from_partial(mut elements: Vec<F>) -> Self {
-        debug_assert!(elements.len() <= 4);
-        while elements.len() < 4 {
-            elements.push(F::ZERO);
-        }
-        Self {
-            elements: [elements[0], elements[1], elements[2], elements[3]],
-        }
+    pub fn from_partial(elements_in: &[F]) -> Self {
+        debug_assert!(elements_in.len() <= 4);
```

Comment:
I don't think this `debug_assert` is necessary anymore: the bounds checking on `elements` should handle it now.

Reply:
Oh nice good to know!